grails-app/controllers/org/pih/warehouse/core/ErrorsController.groovy

```diff
@@ -97,12 +97,12 @@ class ErrorsController {
     def handleValidationErrors() {
         if (RequestUtil.isAjax(request)) {
             response.status = 400
-            BeanPropertyBindingResult errors = request?.exception?.cause?.errors
+            BeanPropertyBindingResult errors = request?.exception?.cause?.target?.errors ?: request?.exception?.cause?.errors
```

Comment:
I'm worried about this change. I would prefer to go backwards and figure out if we can rely on request.exception instead of digging deeper into these exceptions. I think a better approach might be to figure out what kind of exception we have and deciding what to do based on that. 

In other words, if we can be sure that handleValidationErrors() will only be invoked when a ValidationException has been thrown then we should be able to do this
```
request.exception.errors 
```
I don't remember why we decided to jump into the cause. 

If it is necessary for certain cases  then it should probably be done in way that cascades. But I really don't like this approach. 
```
request?.exception?.cause?.target?.errors ?: 
    request?.exception?.cause?.errors ?: 
        request?.exception?.errors
```

Reply:
As I can see `request?.exception` doesn't have either errors or fullMessage property. So, I think that was the reason for jumping into the cause. I tried `request.exception.errors` on validation errors within locations and stock movements and the same situation is here.
![image](https://github.com/openboxes/openboxes/assets/83239466/31fbcb7c-e7a1-43f3-9844-96d152f3ca00)

Reply:
It looks like we are using `cause` since introducing this kind of handling validation error.
![image](https://github.com/openboxes/openboxes/assets/83239466/242c3391-50d8-445c-affc-83bf73b374ce)

Reply:
So in Grails 1 if an exception goes uncaught the GrailsExceptionResolver will be invoked to figure out what to do with it. I believe it uses the UrlMappings with error code "500" to determine where to forward/redirect the request. For ValidationExceptions, we send the request to the handleValidationErrors action of the ErrorsController. Pretty basic. 
```
"500"(controller: "errors", action: "handleException")
"500"(controller: "errors", action: "handleNotFound", exception: ObjectNotFoundException)
"500"(controller: "errors", action: "handleValidationErrors", exception: ValidationException)
"500"(controller: "errors", action: "handleUnauthorized", exception: AuthenticationException)
```
The GrailsExceptionResolver will wrap a runtime exception with a GrailsWrappedRuntimeException which seems to happen in the setStatus method below. 

https://github.com/grails/grails-core/blob/a8336f1f68355e42ada68e3d25f1c5ed23b62b94/grails-web-mvc/src/main/groovy/org/grails/web/errors/GrailsExceptionResolver.java#L87

The resolveException method then forwards to the ErrorsController.handleValidationErrors() method as described above.

https://github.com/grails/grails-core/blob/a8336f1f68355e42ada68e3d25f1c5ed23b62b94/grails-web-mvc/src/main/groovy/org/grails/web/errors/GrailsExceptionResolver.java#L93C18-L93C38

When the error is handled by the ErrorsController it is now a GrailsWrappedRuntimeException, which is why we need to get the "cause". 
```
request.exception.cause
```
I assume this is just calling ExceptionUtils.getRootCause() but haven't had a chance to verify that yet. In any case, cause returns the ValidationException. 

So there should be no reason to dig deeper to get `target` unless we're doing something wrong somewhere. So I guess my confusion is around the where target comes from.

Reply:
My assumption is that `target` is related to the InvocationTargetException which I still don't understand completely. 
```
2023-07-21 01:22:09,780 ERROR [080-exec-3] StackTrace                              : Full Stack Trace:
java.lang.reflect.InvocationTargetException: null
```
I see it in the logs from time to time, but I feel like this is only an issue in development environments. With that said, I don't know for sure.